services/console-proxy/server/src/main/java/com/cloud/consoleproxy/vnc/NoVncClient.java

```diff
@@ -62,6 +69,27 @@ public void connectTo(String host, int port) throws UnknownHostException, IOExce
         setStreams();
     }
 
+    public void connectToWebSocket(String websocketUrl, Session session) throws URISyntaxException {
+        webSocketReverseProxy = new WebSocketReverseProxy(new URI(websocketUrl), session);
+        webSocketReverseProxy.connect();
+    }
+
+    public boolean isVncOverWebSocketConnection() {
+        return webSocketReverseProxy != null;
+    }
+
```

Comment:
Non websocket connections, will cause NPE for the below methods. Either add a comment here to use these methods for _isVncOverWebSocketConnection()_ true, or explicitly add a check.